packages/asset/contracts/AssetCreate.sol

```diff
@@ -0,0 +1,351 @@
+                    tiers,
+                    amounts,
+                    revealed,
+                    metadataHashes
+                )
+            ),
+            "Invalid signature"
+        );
+
+        require(tiers.length == amounts.length, "Arrays must be same length");
+        require(
+            amounts.length == metadataHashes.length,
+            "Arrays must be same length"
+        );
+        require(
+            metadataHashes.length == revealed.length,
+            "Arrays must be same length"
+        );
+
+        uint256[] memory tokenIds = new uint256[](tiers.length);
+        uint256[] memory tiersToBurn = new uint256[](tiers.length);
+        for (uint256 i = 0; i < tiers.length; i++) {
+            uint16 creatorNonce = getCreatorNonce(creator);
+            tiersToBurn[i] = tiers[i];
+            tokenIds[i] = TokenIdUtils.generateTokenId(
+                creator,
+                tiers[i],
+                creatorNonce,
+                revealed[i] ? 1 : 0,
+                false
+            );
+        }
+
+        catalystContract.burnBatchFrom(creator, tiersToBurn, amounts);
+
+        assetContract.mintBatch(creator, tokenIds, amounts, metadataHashes);
+        emit AssetBatchMinted(
+            creator,
+            tokenIds,
+            tiers,
+            amounts,
+            metadataHashes
+        );
+    }
+
+    /// @notice Create special assets, like TSB exclusive tokens
+    /// @dev Only callable by the special minter
+    /// @param signature A signature generated by TSB
+    /// @param tier The tier of the asset to mint
+    /// @param amount The amount of the asset to mint
+    /// @param metadataHash The metadata hash of the asset to mint
+    function createSpecialAsset(
+        bytes memory signature,
+        uint8 tier,
+        uint256 amount,
+        bool revealed,
+        string calldata metadataHash,
+        address creator
+    ) external onlyRole(SPECIAL_MINTER_ROLE) {
+        require(
+            authValidator.verify(
+                signature,
+                _hashMint(
+                    creator,
+                    creatorNonces[creator],
+                    tier,
+                    amount,
+                    revealed,
+                    metadataHash
+                )
+            ),
+            "Invalid signature"
+        );
+
+        uint16 creatorNonce = getCreatorNonce(creator);
+        uint256 tokenId = TokenIdUtils.generateTokenId(
+            creator,
+            tier,
+            creatorNonce,
+            revealed ? 1 : 0,
+            false
+        );
+
+        assetContract.mint(creator, tokenId, amount, metadataHash);
+        emit SpecialAssetMinted(creator, tokenId, amount, metadataHash);
+    }
+
+    /// @notice Get the next available creator nonce
+    /// @param creator The address of the creator
+    /// @return nonce The next available creator nonce
+    function getCreatorNonce(address creator) internal returns (uint16) {
```

Comment:
This function isn't view only so it's not really a 'get' function, since it increments the creatorNonce.
Need to separate "incrementNonce" from "viewNonce" functionality

Reply:
Very good point.
How does `getAndIncrementCreatorNonce` sound?

Reply:
As discussed it's preferable to have get and increment as separate functions